vendor/bat-native-ads/src/bat/ads/internal/eligible_ads/ad_predictor_info.h

```diff
@@ -0,0 +1,32 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_VENDOR_BAT_NATIVE_ADS_SRC_BAT_ADS_INTERNAL_ELIGIBLE_ADS_AD_PREDICTOR_INFO_H_
+#define BRAVE_VENDOR_BAT_NATIVE_ADS_SRC_BAT_ADS_INTERNAL_ELIGIBLE_ADS_AD_PREDICTOR_INFO_H_
+
+#include "bat/ads/internal/segments/segments_aliases.h"
+
+namespace ads {
+
+template <typename T>
+struct AdPredictorInfo {
+  AdPredictorInfo();
+  AdPredictorInfo(const AdPredictorInfo& info);
+  ~AdPredictorInfo();
+
+  T creative_ad;
+  SegmentList segments;
+  bool does_match_intent_child_segments;
+  bool does_match_intent_parent_segments;
```

Comment:
need to add default initalizers for bool members

Reply:
fixed